unittest/abi_tests.cpp

```diff
@@ -0,0 +1,735 @@
+#define BOOST_TEST_MODULE unittests
+#define BOOST_TEST_DYN_LINK
+
+#include <algorithm>
+#include <cstdlib>
+#include <iterator>
+#include <vector>
+
+#include <boost/test/unit_test.hpp>
+
+#include <fc/exception/exception.hpp>
+#include <fc/io/json.hpp>
+#include <fc/log/logger.hpp>
+#include <fc/variant.hpp>
+
+#include <evt/chain/contracts/abi_serializer.hpp>
+#include <evt/chain/contracts/evt_contract.hpp>
+#include <evt/chain/contracts/types.hpp>
+// #include <evt/abi_generator/abi_generator.hpp>
+
+#include <boost/test/framework.hpp>
+
+using namespace evt;
+using namespace chain;
+using namespace contracts;
+
+// BOOST_AUTO_TEST_SUITE(unittests)
+
+// verify that round trip conversion, via bytes, reproduces the exact same data
+fc::variant
+verify_byte_round_trip_conversion(const abi_serializer& abis, const type_name& type, const fc::variant& var) {
+    auto bytes = abis.variant_to_binary(type, var);
+
+    auto var2 = abis.binary_to_variant(type, bytes);
+
+    std::string r = fc::json::to_string(var2);
+
+    auto bytes2 = abis.variant_to_binary(type, var2);
+
+    BOOST_TEST(fc::to_hex(bytes) == fc::to_hex(bytes2));
+
+    return var2;
+}
+
+auto
+get_resolver(const abi_def& abi = abi_def()) {
+    return [&abi](const account_name& name) -> optional<abi_serializer> {
+        return abi_serializer(evt_contract_abi());
```

Comment:
cache abi serializer here, may be a static variable here? don't instantiation every time.